advanced/src/hashTable.js

```diff
@@ -0,0 +1,227 @@
+function Node(key, value) {
+  this.key = key
+  this.value = value
+  this.next = null
+}
+
+
+const LinkedList = class LinkedList {
+  constructor() {
+    this.length = 0
+    this.head = null
+  }
+
+  insert(key, value) {
+    let node = new Node(key, value)
+    let currentNode = this.head
+
+    if (!currentNode) {
+      this.head = node
+      this.length++
+      return node
+    }
+    while (currentNode.next) {
+      currentNode = currentNode.next
+    }
+    currentNode.next = node
+    this.length += 1
+    return node
+  }
+
+//regular find
+  find(key) {
+    var currentNode = this.head
+    if (!currentNode) {
+      return false
+    }
+    while (currentNode) {
+      if (currentNode.key === key){
+        return currentNode
+      }
+    } return false
+  }
+
+//find but return the value
+  findButReturnValue(key) {
+    const node = this.find(key)
+    return node
+      ? node.value
+      : null
+  }
+
+//see if linked list contains key
+  has(key) {
+    return !!this.find(key)
```

Comment:
I'm generally against shortcuts like this in javascript - it relies on some arcane knowledge of the language, rather than making explicit the test you're conducting.  If you're the only developer, this is probably OK, but if you're working on shared codebases, you'll definitely want to lean towards explicit code.